File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int solution(const char* s) {
    int answer = 0;
    int i=0;
    int same = 0;
    int diff = 0;
    char target = ' ';
    while(s[i]){
        if(same == 0){target = s[i];same++;answer++;}
        else{
            if(s[i] == target){same++;}
            else{diff++;}

            if(same == diff){same = 0; diff =0;}
        }
        i++;
    }
    return answer;
}

File: 프로그래머스/1/140108.문자열 나누기-Python/문자열 나누기.py
def solution(s):
    answer = 0

    count_1 = 0
    count_2 = 0
    p = ''
    q = True

    for i in range(len(s)):
        c = s[i]

        if i == len(s) - 1:
            answer += 1
            break

        if q:
            p = c
            q = False
            count_1 += 1
        else:
            if p == c:
                count_1 += 1
            else:
                count_2 += 1
    
            if count_1 == count_2:
                q = True
                answer += 1

    return answer